File: Data Structures & Algorithms/first-missing-positive/submission-0.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int track= 1;
        for(int i= 0; i<nums.size(); i++){
            if(nums[i]<=0){
                continue;
            }

            if(nums[i]==track){
                track++;
            }else if(nums[i]>track){
                return track;
            }
        }

        return track;
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-0.java
class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> ans= new ArrayList<>();
        backtrack(nums,new ArrayList<>(),  ans, 0);
        return ans;
    }

    static void backtrack(int[] nums, List<Integer> subset, List<List<Integer>> ans, int idx){
        
        ans.add(new ArrayList<>(subset));

        for(int i= idx; i<nums.length; i++){
            if(i>idx && nums[i]== nums[i-1]){
                continue;
            }
            subset.add(nums[i]);
            backtrack(nums, subset, ans, i+1);
            subset.remove(subset.size()-1);
        }
    }
}
